### DISCUSSION TITLE
Will there be Aarch64 releases of ccache?

### DISCUSSION THREAD

**Discussion by freddan80** [ccache/ccache#1310] [General]
Hi ccache team,

Are there plans to distribute aarch64 asserts on https://github.com/ccache/ccache/releases/ ?

Cheers,
Fredrik

**Comment by jrosdahl** [comment]
No, there have not been any requests for aarch64 binary releases before.

Would you be interested in working on this? From my perspective, we would need:

* A new CI job that builds ccache with an aarch64 cross compiler.
* A decision on which OS the releases should be built. For reference, x86_64 binary releases are built in a ~CentOS 7~AlmaLinux 8 image in order not to depend on a new libc version. For aarch64, maybe Ubuntu 20.04 would be reasonable?
* Somebody who tests the aarch64 binary releases since I don't have access to an aarch64 system myself.

**Reply by cristianadam** [reply]
I have a Lenovo x13s Arm64 laptop which I can test a Windows 11 Arm64 and Ubuntu 22.04 aarch64 binaries.

**Reply by freddan80** [reply]
Thx @jrosdahl , @cristianadam for the quick reply!

I was mostly curious. One of the project I work with prefers a late ccache version, but I think I can use the distro version for Ubuntu 20.04, v3.7.7. 

> A new CI job that builds ccache with an aarch64 cross compiler.

Do you have arm64 runners in this project?

> A decision on which OS the releases should be built. For reference, x86_64 binary releases are built in a CentOS 7 image in order not to depend on a new libc version. For aarch64, maybe Ubuntu 20.04 would be reasonable?

Ubuntu 20.04 sounds reasonable.

> Somebody who tests the aarch64 binary releases since I don't have access to an aarch64 system myself.

Makes sense. If you had access to an arm64 runner in this project's CI, would that be useful? I.e. is it easy to add build & test jobs to execute on an arm64 runner?

**Reply by jrosdahl** [reply]
> Do you have arm64 runners in this project?

No, we use [the common x86_64-based GitHub-hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

> If you had access to an arm64 runner in this project's CI, would that be useful? I.e. is it easy to add build & test jobs to execute on an arm64 runner?

I guess that would be useful since we could then also run the test suite. Adding new jobs should be fairly easy.

**Comment by freddan80** [comment]
Thx @jrosdahl ! For now we'll use the Ubuntu distro version, but this might bubble up in prio for us at some point.

**Reply by dportok** [reply]
@jrosdahl I am looking to build an aarch64 binary that can run on CentOS 7 aarch64 (just like the corresponding x86_64 binary does) in order not to depend on a new libc version. Unfortunately, I can't find a detailed guide of how this was done for x86_64 so I can try the same for aarch64. Is there a guide somewhere? I have access on aarch64 machine in my job where I will try to execute the build procedure. Thanks in advance

**Reply by jrosdahl** [reply]
I build the Linux x86_64 binary release on ~CentOS 7~AlmaLinux 8 with CMake options `-DCMAKE_BUILD_TYPE=Release -DHIREDIS_FROM_INTERNET=ON -DZSTD_FROM_INTERNET=ON -DCMAKE_EXE_LINKER_FLAGS_INIT=-static-libstdc++`, using GCC from devtoolset-11.

**Reply by dportok** [reply]
Thanks for providing the build options. I will try it and let you know about the result.

**Comment by Kreijstal** [comment]
don't forget to add riscv64 while you are at it too, if it isn't that much work
